src/main/java/com/faforever/api/user/UserService.java

```diff
@@ -124,4 +129,75 @@ void activate(String token) {
     log.debug("User has been activated: {}", user);
     userRepository.save(user);
   }
+
+  void changePassword(String newPassword, User user) {
+    user.setPassword(passwordEncoder.encode(newPassword));
+    userRepository.save(user);
+  }
+
+  void changeLogin(String newLogin, User user) {
+    user.setLogin(newLogin);
+    userRepository.save(user);
+  }
+
+  void resetPassword(String email) {
+    log.debug("Registration requested for user: {}", email);
+
+    User user = userRepository.findOneByEmailIgnoreCase(email);
+
+    if (user == null) {
+      throw new ApiException(new Error(ErrorCode.USERNAME_INVALID));
+    }
+
+    String token = createPasswordResetToken(user.getId());
+    String passwordResetUrl = String.format(properties.getPasswordReset().getPasswordResetUrlFormat(), token);
+
+    emailService.sendPasswordResetMail(user.getLogin(), user.getEmail(), passwordResetUrl);
+  }
+
+  @SneakyThrows
+  private String createPasswordResetToken(int userId) {
+    int expirationSeconds = properties.getRegistration().getLinkExpirationSeconds();
+
+    String claim = objectMapper.writeValueAsString(ImmutableMap.of(
+        KEY_ACTION, ACTION_RESET_PASSWORD,
+        KEY_EXPIRY, Instant.now().plusSeconds(expirationSeconds).toString(),
+        KEY_USER_ID, userId
+    ));
+
+    return JwtHelper.encode(claim, macSigner).getEncoded();
+  }
+
+  @SneakyThrows
+  void claimPasswordResetToken(String token, String newPassword) {
+    HashMap<String, String> claims = objectMapper.readValue(JwtHelper.decodeAndVerify(token, macSigner).getClaims(), HashMap.class);
+
+    String action = claims.get(KEY_ACTION);
+    if (!Objects.equals(action, ACTION_RESET_PASSWORD)) {
+      throw new ApiException(new Error(ErrorCode.TOKEN_INVALID));
+    }
+    if (Instant.parse(claims.get(KEY_EXPIRY)).isBefore(Instant.now())) {
+      throw new ApiException(new Error(ErrorCode.TOKEN_EXPIRED));
+    }
+
+    int userId = Integer.parseInt(claims.get(KEY_USER_ID));
+    User user = userRepository.findOne(userId);
+
+    if (user == null) {
+      throw new ApiException(new Error(ErrorCode.TOKEN_INVALID));
+    }
+
+    user.setPassword(passwordEncoder.encode(newPassword));
+    userRepository.save(user);
+  }
+
+  public User getUser(Authentication authentication) {
```

Comment:
This should be moved in the `UserService`, similiar to https://github.com/FAForever/faf-java-api/blob/develop/src/main/java/com/faforever/api/player/PlayerService.java#L20